documentation/synthetic_tokens/security/security.md

```diff
@@ -0,0 +1,107 @@
+We’ve categorized and ranked the open issues by likelihood of impact. 
+
+Note that the financial contract template has an emergency shutdown feature that prematurely expires the contract. 
+This can only be triggered by a vote approved by UMA token holders. 
+Emergency shutdown prematurely expires the contract and the final resolution depends on a DVM price, which could take 2-4 days to arrive.
+
+## Open issues affecting incentives to liquidate:
+### No haircut mechanisms implemented
+* Scenario: If any token sponsor’s position is collateralized by < 100%, there is no incentive to liquidate them. 
+If the contract expires with any sponsors below 100% collateral, there will be a race to pull collateral from the contract. 
+Whichever sponsors and token holders redeem first will receive collateral. 
+Whoever comes last will receive less or nothing depending on the amount of collateral left in the contract.
+* Solution: Although we expect the economic incentives to work without a haircut mechanism, if a position does reach < 100% collateralization, UMA will liquidate these positions uneconomically.
+
+## Open issues affecting capital efficiency: 
+### New token creation
+* Scenario: Token sponsors who wish to create new tokens are required to collateralize the new tokens at levels that meet or exceed the global collateralization ratio (GCR). 
+However, this makes the role of token sponsor capital inefficient for 2 reasons: 
+1. If a token sponsor has an outstanding synthetic token position collateralized well above the GCR, such that the creation of new tokens could be supported by the current amount of excess collateral in their position, they cannot use this collateral to create new synthetic tokens. 
+1. If a token sponsor wishes to create new tokens collateralized to a level below the GCR, they must first collateralize their position above the GCR and go through a “slow” withdrawal to withdraw their excess collateral.
+
+### Whale griefing attack
+* Scenario: A griefing sponsor can add a proportionally large amount of collateral relative to the rest of the contract while creating a very small debt position. 
+This drastically imbalances the contract GCR upwards. 
+For example, a malicious sponsor can add 100x the current contract collateral while only creating a 1wei position in debt. 
+This results in the contract GCR drastically increasing (~100x), making all later sponsors required to be massively over collateralized. 
+At best, this slows down future token sponsorship (must repeatedly create and wait on withdraw), and at worst, this can lock all future sponsorship, constricting token supply. 
+This could make it difficult or impossible for liquidators to liquidate positions and simultaneously drive up the token price.
+One worrisome attack vector is if a sponsor were to use this to constrict token supply, and then issue a withdraw request to bring their collateralization ratio to 0%. 
```

Comment:
this is phrased awkwardly—like it's a second attack vector.

Reply:
Would you prefer if we rephrased "attack vector" to "edge case"?